org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.code.neighborhood;
+
+import java.io.File;
+import java.io.FileInputStream;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class GridFactory {
+  private static final String GRID_FILE_NAME = "grid.txt";
+  private static final String GRID_SQUARE_TYPE_FIELD = "tileType";
+  private static final String GRID_SQUARE_VALUE_FIELD = "value";
+
+  protected Grid createGridFromJSON(String filename) {
+    File file = new File(GRID_FILE_NAME);
+    FileInputStream fis;
+    try {
+      fis = new FileInputStream(file);
+      byte[] data = new byte[(int) file.length()];
+      fis.read(data);
+      fis.close();
+      return createGridFromString(new String(data, "UTF-8"));
+    } catch (Exception e) {
+      throw new UnsupportedOperationException("Try adding a grid.txt file.");
+    }
+  }
+
+  // Creates a grid from a string, assuming that the string is a 2D JSONArray of JSONObjects
+  // with each JSONObject containing an integer tileType and optionally an integer value
+  // corresponding with the paintCount for that tile.
+  protected Grid createGridFromString(String description) {
+    String[] lines = description.split("\n");
+    String parseLine = String.join("", lines);
+    JSONParser jsonParser = new JSONParser();
+    try {
+      Object obj = jsonParser.parse(parseLine);
+      JSONArray gridSquares = (JSONArray) obj;
+      int height = gridSquares.size();
+      if (height == 0) {
+        throw new UnsupportedOperationException("Please check the format of your grid.txt file");
+      }
+      int width = ((JSONArray) gridSquares.get(0)).size();
```

Comment:
We'll want to add a check here that the grid is square (when this was first written, that wasn't a requirement, but has since been clarified)